server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+        """
+
+        mock_repo = {
+            'id': 'mock_repo', 'display_name': 'mock_display', 'description': 'mock_desciption',
+            'notes': 'mock_notes', 'importer_type_id': 'mock_importer', 'distributors': ['dist1']
+        }
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps(mock_repo)
+        mock_reverse.return_value = '/mock/path/'
+        mock_repo_manager = mock_factory.repo_manager.return_value
+        mock_repo_manager.create_and_configure_repo.return_value = mock_repo
+
+        repos_view = ReposView()
+        response = repos_view.post(mock_request)
+
+        mock_repo['_href'] = '/mock/path/'
+        mock_resp.assert_called_once_with(mock_repo)
+        mock_redir.assert_called_once_with(mock_resp.return_value, mock_repo['_href'])
+        self.assertTrue(response is mock_redir.return_value)
+
+
+class TestRepoResourceView(unittest.TestCase):
+    """
+    Tests for RepoResoureceView.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_get_existing_repo(self, mock_factory, mock_resp):
+        """
+        Retrieve an existing repository.
+        """
+
+        mock_repo = {'mock_repo': 'somedata'}
+        mock_q_manager = mock_factory.repo_query_manager.return_value
+        mock_q_manager.find_by_id.return_value = mock_repo
+        mock_request = mock.MagicMock()
+        mock_request.GET = {}
+
+        repos_resource = RepoResourceView()
+        response = repos_resource.get(mock_request, 'mock_repo')
+
+        mock_resp.assert_called_once_with(mock_repo)
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_get_nonexisting_repo(self, mock_factory, mock_resp):
+        """
+        Retrieve a nonexisting repository.
+        """
+
+        mock_q_manager = mock_factory.repo_query_manager.return_value
+        mock_q_manager.find_by_id.return_value = None
+        mock_request = mock.MagicMock()
+
+        repos_resource = RepoResourceView()
+        try:
+            repos_resource.get(mock_request, 'mock_repo')
+        except pulp_exceptions.MissingResource, response:
+            pass
+        else:
+            raise AssertionError("MissingResource should be raised for a nonexisting repository")
+
+        self.assertEqual(response.http_status_code, 404)
+        self.assertTrue(response.error_code is error_codes.PLP0009)
```

Comment:
you could also add `self.assertEqual(response.error_data['resources'], {'repo': 'mock_repo'})`